src/vm/internal/otable.cc

```diff
@@ -1,271 +1,148 @@
+        }
+        return true;
+    }
+    else {
+        return false;
+    }
 }
 
-/*
- * Lookup an object in the hash table; if it isn't there, return null.
- * This is only different to find_object_n in that it collects different
- * stats; more finds are actually done than the user ever asks for.
- */
-
-static long user_obj_lookups = 0, user_obj_found = 0;
-
-object_t *lookup_object_hash(const char *s) {
-  object_t *ob = find_obj_n(s);
+//write some information about memory usage and other statistics of the object table to out_buffer_t.
+//TODO: remove dependency on outbuffer_t here
+#ifndef TESTING
+int ObjectTable::showStatus(outbuffer_t *out, int verbose) {
+
+    std::stringstream ss;
+    if (verbose == 1) {
+        ss <<  "Object name hash table status:" << std::endl;
+        ss <<  "Object name hash table status:" << std::endl;
+        ss << "------------------------------" << std::endl;
+        ss << "Elements:        " << objects_.size() << std::endl;
+        ss << "Memory(bytes):     " << objects_.size() * sizeof(Value) << std::endl;
+        ss << "Bucket count:    " << objects_.bucket_count() << std::endl;
+        ss << "Load factor:     " << objects_.load_factor() << std::endl;
+
+        outbuf_add(out, ss.str().c_str());
+      }
 
-  user_obj_lookups++;
-  if (ob) {
-    user_obj_found++;
-  }
-  return (ob);
+    if (!verbose) {
+        ss << "Memory used(bytes):     " << objects_.size() * sizeof(Value) << std::endl;
+        outbuf_add(out, ss.str().c_str());
+    }
+    return objects_.size() * sizeof(Value);
 }
+#endif
 
-/*
- * Print stats, returns the total size of the object table.  All objects
- * are in table, so their size is included as well.
- */
-
-static char sbuf[100];
-
-int show_otable_status(outbuffer_t *out, int verbose) {
-  const auto object_hash_table_size = CONFIG_INT(__OBJECT_HASH_TABLE_SIZE__);
-
-  int starts;
-
-  if (verbose == 1) {
-    outbuf_add(out, "Object name hash table status:\n");
-    outbuf_add(out, "------------------------------\n");
-    sprintf(sbuf, "%10.2f", objs_in_table / static_cast<float>(object_hash_table_size));
-    outbuf_addv(out, "Average hash chain length:       %s\n", sbuf);
-    sprintf(sbuf, "%10.2f", static_cast<float>(obj_probes) / obj_searches);
-    outbuf_addv(out, "Average search length:           %s\n", sbuf);
-    outbuf_addv(out, "Internal lookups (succeeded):    %lu (%lu)\n",
-                obj_searches - user_obj_lookups, objs_found - user_obj_found);
-    outbuf_addv(out, "External lookups (succeeded):    %lu (%lu)\n", user_obj_lookups,
-                user_obj_found);
-  }
-  starts = static_cast<long>(object_hash_table_size) * sizeof(object_t *) +
-           objs_in_table * sizeof(object_t);
-
-  if (!verbose) {
-    outbuf_addv(out, "Obj table overhead:\t\t%8d %8d\n",
-                object_hash_table_size * sizeof(object_t *), starts);
-  }
-  return starts;
-}
+std::string basename(std::string s) {
+    
```

Comment:
no leading blank line

Reply:
I think without the leading space the comment is harder to read.

Reply:
It's not for debate, it's what the styleguide and rest of the code's style.